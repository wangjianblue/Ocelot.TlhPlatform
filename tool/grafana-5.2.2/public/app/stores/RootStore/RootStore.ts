import { types } from 'mobx-state-tree';
import { SearchStore } from './../SearchStore/SearchStore';
import { ServerStatsStore } from './../ServerStatsStore/ServerStatsStore';
import { NavStore } from './../NavStore/NavStore';
import { AlertListStore } from './../AlertListStore/AlertListStore';
import { ViewStore } from './../ViewStore/ViewStore';
import { FolderStore } from './../FolderStore/FolderStore';
import { PermissionsStore } from './../PermissionsStore/PermissionsStore';

export const RootStore = types.model({
  search: types.optional(SearchStore, {
    sections: [],
  }),
  serverStats: types.optional(ServerStatsStore, {
    stats: [],
  }),
  nav: types.optional(NavStore, {}),
  alertList: types.optional(AlertListStore, {
    rules: [],
  }),
  permissions: types.optional(PermissionsStore, {
    fetching: false,
    items: [],
  }),
  view: types.optional(ViewStore, {
    path: '',
    query: {},
    routeParams: {},
  }),
  folder: types.optional(FolderStore, {}),
});

type IRootStoreType = typeof RootStore.Type;
export interface IRootStore extends IRootStoreType {}
