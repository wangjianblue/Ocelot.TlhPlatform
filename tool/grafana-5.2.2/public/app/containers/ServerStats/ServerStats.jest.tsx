import React from 'react';
import renderer from 'react-test-renderer';
import { ServerStats } from './ServerStats';
import { RootStore } from 'app/stores/RootStore/RootStore';
import { backendSrv, createNavTree } from 'test/mocks/common';

describe('ServerStats', () => {
  it('Should render table with stats', done => {
    backendSrv.get.mockReturnValue(
      Promise.resolve({
        dashboards: 10,
      })
    );

    const store = RootStore.create(
      {},
      {
        backendSrv: backendSrv,
        navTree: createNavTree('cfg', 'admin', 'server-stats'),
      }
    );

    const page = renderer.create(<ServerStats backendSrv={backendSrv} {...store} />);

    setTimeout(() => {
      expect(page.toJSON()).toMatchSnapshot();
      done();
    });
  });
});
