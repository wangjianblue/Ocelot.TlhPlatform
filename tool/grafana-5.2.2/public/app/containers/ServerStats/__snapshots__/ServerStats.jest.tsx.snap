// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServerStats Should render table with stats 1`] = `
<div>
  <div
    className="page-header-canvas"
  >
    <div
      className="page-container"
    >
      <div
        className="page-header"
      >
        <div
          className="page-header__inner"
        >
          <span
            className="page-header__logo"
          >
            
            
          </span>
          <div
            className="page-header__info-block"
          >
            <h1
              className="page-header__title"
            >
              admin-Text
            </h1>
            
          </div>
        </div>
        <nav>
          <div
            className="gf-form-select-wrapper width-20 page-header__select-nav"
          >
            <label
              className="gf-form-select-icon "
              htmlFor="page-header-select-nav"
            />
            <select
              className="gf-select-nav gf-form-input"
              id="page-header-select-nav"
              onChange={[Function]}
              value="/url/server-stats"
            >
              <option
                value="/url/server-stats"
              >
                server-stats-Text
              </option>
            </select>
          </div>
          <ul
            className="gf-tabs page-header__tabs"
          >
            <li
              className="gf-tabs-item"
            >
              <a
                className="gf-tabs-link active"
                href="/url/server-stats"
                target={undefined}
              >
                <i
                  className=""
                />
                server-stats-Text
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </div>
  <div
    className="page-container page-body"
  >
    <table
      className="filter-table form-inline"
    >
      <thead>
        <tr>
          <th>
            Name
          </th>
          <th>
            Value
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            Total dashboards
          </td>
          <td>
            10
          </td>
        </tr>
        <tr>
          <td>
            Total users
          </td>
          <td>
            0
          </td>
        </tr>
        <tr>
          <td>
            Active users (seen last 30 days)
          </td>
          <td>
            0
          </td>
        </tr>
        <tr>
          <td>
            Total orgs
          </td>
          <td>
            0
          </td>
        </tr>
        <tr>
          <td>
            Total playlists
          </td>
          <td>
            0
          </td>
        </tr>
        <tr>
          <td>
            Total snapshots
          </td>
          <td>
            0
          </td>
        </tr>
        <tr>
          <td>
            Total dashboard tags
          </td>
          <td>
            0
          </td>
        </tr>
        <tr>
          <td>
            Total starred dashboards
          </td>
          <td>
            0
          </td>
        </tr>
        <tr>
          <td>
            Total alerts
          </td>
          <td>
            0
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
